### ISSUE TITLE
API Documentaton

### ISSUE THREAD

**Issue by PythonNut** [myuhe/smartrep.el#15]
Hello,

First, I'd like to say thank you. This is a tremendously helpful piece of code you have here, and It's significantly simplified my emacs life. I don't use it much for repetition, rather, I hijack `smartrep-read-event-loop` to create a bunch of cool commands.

In short, there's more to this code that you advertise. The ability to do something, and then _dynamically_ decide whether to be a prefix command or not, is amazing. Combined with the fact that the event loop is easily broken programatically, and that these loops can, at any time, start other loops is mindblowing.

In general, it would be great if your summary and readme.org could reflect this. I suspect that these features are lost on almost everyone. (In fact, I suspect that everything is lost, since the readme is empty, and there is no substantial in-code documentation either).

I would even go so far as to suggest pulling out the prefix-map stuff into it's own library. I suspect that a lot of projects could benefit from it. Here's a short, informal a list of projects that have their own version of this functionality.

  * [expand-region](https://github.com/magnars/expand-region.el)
  * [jump-char](https://github.com/lewang/jump-char)
  * [easy-kill](https://github.com/leoliu/easy-kill)
  * cua-mode

In short, awesome project. Document it. :)
